### DISCUSSION TITLE
DXFTypeError: name has to be a string.

### DISCUSSION THREAD

**Discussion by Gianmaria92** [mozman/ezdxf#1106] [Q&A]
Good morning, 
when I try to open a .dxf in python, I get the error ‘DXFTypeError: name has to be a string.’
The .dxf file is generated by c#, and is readable by standard software such as DraftSight for example. 
What am I missing?

**Comment by mozman** [comment]
That's not enough information...

**Reply by Gianmaria92** [reply]
[tang_1000.zip](https://github.com/user-attachments/files/15787610/tang_1000.zip)
Please find the file attached, I try to open it as simple as in the guide: 

```
import ezdxf
import matplotlib.pyplot as plt
from matplotlib.backends.backend_agg import FigureCanvasAgg as FigureCanvas
from ezdxf import recover

def convert_dxf_to_tiff(input_directory, output_directory):
    if not os.path.exists(output_directory):
        os.makedirs(output_directory)

    for filename in os.listdir(input_directory):
        if filename.endswith(".dxf"):
            filepath = os.path.join(input_directory, filename)
            try:
                doc = ezdxf.readfile(filepath)
```

**Comment by mozman** [comment]
This file is a DXF R12 file which  includes structure elements from later DXF formats, this confuses `ezdxf`, maybe I can add this to the `recover` module.

```
VPORT
5
1
70
1
0
VPORT
5
2
100        <<< DXF R13 and later
AcDbSymbolTableRecord           <<< DXF R13 and later
2
*ACTIVE
70
0
```
Although AutoCAD can read this file, it is still a sign of a poor DXF eporter.

**Comment by mozman** [comment]
I fixed this issue and added your example file "tang_1000.dxf" as a test case. I hope this is ok for you. I don't think this is confidential data:

![image](https://github.com/mozman/ezdxf/assets/4230966/9ca33b12-dfce-47c5-9381-e3a2cc73680f)

**Reply by Gianmaria92** [reply]
That's great, thanks. Ok for me. 
What can I do now to read those kind of files? 
Thanks

**Reply by mozman** [reply]
You have to use the recover module:

```python
from ezdxf import recover

doc, auditor = recover.readfile("your.dxf")
```
Here are the docs for the recover module: https://ezdxf.mozman.at/docs/drawing/recover.html
